### PR TITLE
Nft test pre

### LINKED ISSUE #70: nft pallet: Event and run_to_block
**Description**
finish the test basic implementation regarding events. Under nft pallet, mock.rs/test.rs will be influenced.

**Acceptance Criteria**
- [ ] Add frame system and pallet balance events as well.
- [ ] runing_to_block to finialize the event log; safety check regarding blockNumber manipulation

### PR DESCRIPTION
**Pull request by wangminqi**
This is PR is for issue #70 .
Only few modifies to demonstrate the solution template for event test and run_in_block blockNumber consistence.

### CODE REVIEW

File: pallets/nft/src/tests.rs [hunk 1]
```diff
@@ -56,9 +58,12 @@ fn test_issue_and_claim_eth() {
 		]];
 
 		run_to_block(1);
+		assert_eq!(System::block_number(),1);
```
Comment (Satoshi-Kusumoto) [inline pallets/nft/src/tests.rs]:
```suggestion
		assert_eq!(System::block_number(), 1);
```

File: pallets/nft/src/tests.rs [hunk 2]
```diff
@@ -69,13 +74,17 @@ fn test_issue_and_claim_eth() {
...
+		// even we clear events before create_class, as Pot address never exists and balance on it being added, it will always trigger system event and balance event first time
+		// Improvement will be made in full test - issue #71
+		assert_eq!(
+			events(),
+			[
+			SystemEvent::NewAccount(Pot::get()).into(),
+			BalanceEvent::Endowed(Pot::get(), CREATION_FEE.into()).into(),
+			BalanceEvent::Transfer(alice_account.clone(), Pot::get(), CREATION_FEE.into()).into(),
+			Event::Nft(crate::Event::CreatedClass(alice_account.clone(),0)),
+			]
```
Comment (Satoshi-Kusumoto) [inline pallets/nft/src/tests.rs]:
```suggestion
			[
				SystemEvent::NewAccount(Pot::get()).into(),
				BalanceEvent::Endowed(Pot::get(), CREATION_FEE.into()).into(),
				BalanceEvent::Transfer(alice_account.clone(), Pot::get(), CREATION_FEE.into()).into(),
				Event::Nft(crate::Event::CreatedClass(alice_account.clone(), 0)),
			]
```

**Review by Satoshi-Kusumoto — CHANGES_REQUESTED** [review]

**Comment by geastwood** [comment]
Does Rust comes with a linting tool that we can build into CI? @Satoshi-Kusumoto

**Comment by Satoshi-Kusumoto** [comment]
> Does Rust comes with a linting tool that we can build into CI? @Satoshi-Kusumoto

We are already using rust fmt, @h4n0 is there an issue to put this into CI? I cann't find it

File: pallets/nft/src/mock.rs [hunk 3]
```diff
@@ -118,7 +118,7 @@ pub fn run_to_block(n: u32) {
 		System::set_block_number(System::block_number() + 1);
 		<System as OnInitialize<u32>>::on_initialize(System::block_number());
 		<Nft as OnInitialize<u32>>::on_initialize(System::block_number());
-	}
+	};
```
Comment (Satoshi-Kusumoto) [inline pallets/nft/src/mock.rs]:
```suggestion
	}
```

File: pallets/nft/src/tests.rs [hunk 4]
```diff
@@ -69,13 +74,17 @@ fn test_issue_and_claim_eth() {
...
-		// 	events(),
-		// 	[Event::Nft(crate::Event::CreatedClass(
-		// 		alice_account.clone(),
-		// 		0
-		// 	)),]
-		// );
+		// even we clear events before create_class, as Pot address never exists and balance on it being added, it will always trigger system event and balance event first time
+		// Improvement will be made in full test - issue #71
+		assert_eq!(
+			events(),
```
Comment (Satoshi-Kusumoto) [inline pallets/nft/src/tests.rs:80]: As we have discussed, we should use index instead of compare the whole event array

**Review by Satoshi-Kusumoto — CHANGES_REQUESTED** [review]

**Comment by Satoshi-Kusumoto** [comment]
Also you should use key words like "close" or "fix" otherwise the issue cann't be auto closed

File: pallets/nft/src/tests.rs (hunk 4 above)
Comment (wangminqi) [reply pallets/nft/src/tests.rs:80]: OK. I will rearrange this part into this PR.

**Pull request closed without merging**